src/components/Hoverable/index.js

```diff
@@ -1,189 +1,181 @@
+
+            setIsHovered(false);
+        },
+        [isHovered],
+    );
+
+    useEffect(() => {
+        if (!DeviceCapabilities.hasHoverSupport()) {
             return;
         }
 
-        if (this.wrapperView.contains(e.target)) {
-            return;
-        }
+        document.addEventListener('mouseover', unsetHoveredIfOutside);
 
-        this.setIsHovered(false);
-    }
+        return () => document.removeEventListener('mouseover', unsetHoveredIfOutside);
+    }, [unsetHoveredIfOutside]);
 
-    handleVisibilityChange() {
-        if (document.visibilityState !== 'hidden') {
+    useEffect(() => {
+        if (!disabled || !isHovered) {
             return;
         }
+        setIsHovered(false);
+    }, [disabled, isHovered]);
 
-        this.setIsHovered(false);
-    }
-
-    render() {
-        let child = this.props.children;
-        if (_.isArray(this.props.children) && this.props.children.length === 1) {
-            child = this.props.children[0];
+    useEffect(() => {
+        if (disabled) {
+            return;
         }
-
-        if (_.isFunction(child)) {
-            child = child(this.state.isHovered);
+        if (onHoverIn && isHovered) {
+            return onHoverIn();
         }
-
-        if (!DeviceCapabilities.hasHoverSupport()) {
-            return child;
+        if (onHoverOut && !isHovered) {
+            return onHoverOut();
         }
-
-        return React.cloneElement(React.Children.only(child), {
-            ref: (el) => {
-                this.wrapperView = el;
-
-                // Call the original ref, if any
-                const {ref} = child;
-                if (_.isFunction(ref)) {
-                    ref(el);
-                    return;
-                }
-
-                if (_.isObject(ref)) {
-                    ref.current = el;
-                }
-            },
-            onMouseEnter: (el) => {
-                this.setIsHovered(true);
-
-                if (_.isFunction(child.props.onMouseEnter)) {
-                    child.props.onMouseEnter(el);
-                }
-            },
-            onMouseLeave: (el) => {
-                this.setIsHovered(false);
-
-                if (_.isFunction(child.props.onMouseLeave)) {
-                    child.props.onMouseLeave(el);
-                }
-            },
-            onBlur: (el) => {
-                // Check if the blur event occurred due to clicking outside the element
-                // and the wrapperView contains the element that caused the blur and reset isHovered
-                if (!this.wrapperView.contains(el.target) && !this.wrapperView.contains(el.relatedTarget)) {
-                    this.setIsHovered(false);
-                }
-
-                if (_.isFunction(child.props.onBlur)) {
-                    child.props.onBlur(el);
-                }
-            },
-        });
+    }, [disabled, isHovered, onHoverIn, onHoverOut]);
+
+    useImperativeHandle(outerRef, () => ref.current, []);
```

Comment:
Can we add a comment why this is necessary please?

Reply:
it looks like this is basically exposing the InnerHoverable ref in Hoverable - if that's true it would be great to just leave a comment here to clarify for anyone else who comes across this :D

Reply:
Thanks for the suggestion @Beamanator! Done ✅

Reply:
Thanks!